zerver/views/webhooks/librato.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import absolute_import
+
+import itertools
+
+from django.utils.datetime_safe import datetime
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.response import json_success, json_error
+from zerver.models import get_client
+from zerver.lib.actions import check_send_message
+
+import ujson
+
+
+@api_key_only_webhook_view
+def api_librato_webhook(request, user_profile):
+    def parse_alert(payload):
+        subject = "Alerts"
+        trigger_time = datetime.fromtimestamp((payload['trigger_time']))
+        alert = payload['alert']
+        runbook_url = alert['runbook_url']
+        description = alert['description']
+        alert_id = alert['id']
+        alert_name = alert['name']
+        content = "**Alert #{alert_id} {alert_name} occured at {trigger_time}!**    \n".format(alert_id=alert_id,
+                                                                                               alert_name=alert_name,
+                                                                                               trigger_time=trigger_time)
+        content += "{description}\n".format(description=description)
+        content += "Reaction steps: {runbook_url}\n".format(runbook_url=runbook_url)
+        content += parse_condition_and_violation(payload)
+        return subject, content
+
+    def parse_condition_and_violation(payload):
+        conditions = payload['conditions']
+        violations = payload['violations']['test-source']
+        content = "**Status:**\n"
+        for condition, violation in itertools.izip(conditions, violations):
+            duration = condition['duration']
+            summary_function = condition['summary_function']
+            threshold = condition.get('threshold', '')
+            condition_type = condition['type']
+            metric_name = violation['metric']
+            condition_violated = violation['condition_violated']
+            value = violation['value']
+            recorded_at = datetime.fromtimestamp((violation['recorded_at']))
+            content += "Metric: {metric_name}, {summary_function} was {condition_type} {threshold} by {duration}s\n".format(
+                    metric_name=metric_name, summary_function=summary_function, condition_type=condition_type,
+                    duration=duration, threshold=threshold)
+            content += "Actual value is {value}, recorded at {recorded_at}\n\n".format(value=value,
+                                                                                       condition_violated=condition_violated,
+                                                                                       recorded_at=recorded_at)
+        return content
```

Comment:
Is there a way that we can include a unique identifier for which alert type was triggered in the topic of the message, similar to what we do with the Nagios alert?  I think that would make the threading of the messages better.